### PR TITLE
[10.0][MIG] stock_picking_transfer_lot_autoassign: Migration to 10.0

### PR DESCRIPTION
**Pull request by rgarnau**
cc @jbeficent @etobella

### CODE REVIEW

File: stock_picking_transfer_lot_autoassign/models/stock_picking.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+# Copyright 2017 Pedro M. Baeza <pedro.baeza@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class StockPicking(models.Model):
+    _inherit = 'stock.picking'
+
+    @api.model
```
Comment (rousseldenis) [inline stock_picking_transfer_lot_autoassign/models/stock_picking.py]: Even if it was done like this in v9 module, we cannot do that as there is an inconsistency between self value (which is a single recordset) and the 'model' decorator.

I think we don't have to decorate (like in stock module)

File: stock_picking_transfer_lot_autoassign/wizards/stock_backorder_confirmation.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+class StockBackorderConfirmation(models.TransientModel):
+    _inherit = 'stock.backorder.confirmation'
+
+    @api.multi
+    def process_assign_ordered_qty(self):
+        """Autocomplete product qty with producto ordered qty only for
+        products without lots track
+        """
+        self.ensure_one()
+        for pack_operation in self.pick_id.pack_operation_ids:
```
Comment (rousseldenis) [inline stock_picking_transfer_lot_autoassign/wizards/stock_backorder_confirmation.py]: `for pack_operation in self.pick_id.pack_operation_ids.filtered(lambda o: not o.pack_lot_ids):`

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review. Minor comments

File: stock_picking_transfer_lot_autoassign/models/stock_picking.py (hunk 1 above)
Comment (pedrobaeza) [reply stock_picking_transfer_lot_autoassign/models/stock_picking.py]: This has been perverted in the migration. In v9 the decorator has all the sense due to standard signature:

https://github.com/odoo/odoo/blob/7eb7e9fd6e72764f28df19ab8803e1cceff38b4d/addons/stock/stock.py#L1204

And the inherited method adapted to new API:

https://github.com/OCA/stock-logistics-workflow/blob/c466ffc62170b6b28f059e6ad6fba85624f8bbaf/stock_picking_transfer_lot_autoassign/models/stock_picking.py#L12

with `picking` as the argument and without using `self`

Comment (rousseldenis) [reply stock_picking_transfer_lot_autoassign/models/stock_picking.py]: Ok, so no decorator here.

Comment (pedrobaeza) [reply stock_picking_transfer_lot_autoassign/models/stock_picking.py]: Yes, as `@api.multi` is now the default.

Comment (rousseldenis) [reply stock_picking_transfer_lot_autoassign/models/stock_picking.py]: Be careful that in the super, Odoo consider self as a single recordset

**Comment by rgarnau** [comment]
Thanks @rousseldenis - I've introduced the changes.

**Review by rousseldenis — APPROVED** [review]

File: stock_picking_transfer_lot_autoassign/models/stock_picking.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Pedro M. Baeza <pedro.baeza@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
```
Comment (rousseldenis) [inline stock_picking_transfer_lot_autoassign/models/stock_picking.py]: @rgarnau Sorry, missed this : remove api

Comment (rgarnau) [reply stock_picking_transfer_lot_autoassign/models/stock_picking.py]: done :+1:

File: stock_picking_transfer_lot_autoassign/README.rst [hunk 4]
```diff
@@ -1,5 +1,5 @@
 .. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
-   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :target: http://www.gnu.org/licenses/agpl
```
Comment (JordiBForgeFlow) [inline stock_picking_transfer_lot_autoassign/README.rst]: replace with https

File: stock_picking_transfer_lot_autoassign/wizards/stock_backorder_confirmation.py [hunk 5]
```diff
@@ -2,20 +2,18 @@
...
 
-from openerp import api, models
+from odoo import api, models
 
 
 class StockBackorderConfirmation(models.TransientModel):
     _inherit = 'stock.backorder.confirmation'
 
     @api.multi
     def process_assign_ordered_qty(self):
```
Comment (JordiBForgeFlow) [inline stock_picking_transfer_lot_autoassign/wizards/stock_backorder_confirmation.py]: @sergio-teruel What is the purpose of this feature? I do not understand it. To me the backorder wizard is not the place to assign automatically all quantities to the existing picking. In any case the new module `stock_pack_operation_auto_fill` does what you seem to require more elegantly. https://github.com/OCA/stock-logistics-workflow/pull/361

I propose to remove this feature from the current module.

**Review by JordiBForgeFlow — CHANGES_REQUESTED** [review]
This module works best when used in combination of stock_pack_operation_auto_fill. Not sure if both modules should be merged together.

File: stock_picking_transfer_lot_autoassign/wizards/stock_backorder_confirmation.py (hunk 5 above)
Comment (JordiBForgeFlow) [reply stock_picking_transfer_lot_autoassign/wizards/stock_backorder_confirmation.py]: @sergio-teruel what do you think about this?

**Comment by rousseldenis** [comment]
@rgarnau What is the status of this ?

**Comment by rgarnau** [comment]
@rousseldenis - Migrating this module we saw that it had a feature that from our point of view, is not really related to the lot autoassign purpose and it is in another module already. That's why we are waiting for @sergio-teruel to check if it's really like that.
What's your opinion about it?

**Comment by rousseldenis** [comment]
@rgarnau Yes, stock_pack_operation_auto_fill will populate operations with lot if I remember

**Comment by rgarnau** [comment]
@jbeficent - attended review. The StockBackorderConfirmation has been removed as it is no longer useful.

**Review by JordiBForgeFlow — APPROVED** [review]

**Comment by JordiBForgeFlow** [comment]
@rousseldenis @Cedric-Pigeon This one looks good to me. Can you check from your side?

**Pull request merged**